src/Stack/Config.hs

```diff
@@ -146,9 +146,17 @@ configFromConfigMonoid configStackRoot mproject configMonoid@ConfigMonoid{..} =
                 return $ progsDir </> $(mkRelDir stackProgName) </> platform
             _ -> return $ configStackRoot </> $(mkRelDir "programs") </> platform
 
-     configLocalBin <- do
-        localDir <- liftIO (getAppUserDataDirectory "local") >>= parseAbsDir
-        return $ localDir </> $(mkRelDir "bin")
+     configLocalBin <- case configMonoidLocalBin of
+       Nothing -> do
+           localDir <- liftIO (getAppUserDataDirectory "local") >>= parseAbsDir
+           return $ localDir </> $(mkRelDir "bin")
+       Just userPath -> do
+           tryPath <- try (liftIO $ canonicalizePath userPath >>= parseAbsDir)
+           case tryPath of
+               Left (_ :: SomeException) ->
```

Comment:
This shouldn't be a `SomeException`, but an `IOException`. Also, displaying that information to the user would be a good thing.